src/dune_rules/module_compilation.ml

```diff
@@ -363,19 +363,22 @@ module Alias_module = struct
     }
 
   type t =
-    { main_module : Module_name.t
+    { main_module : Module_name.t option
     ; aliases : alias list
     ; shadowed : Module_name.t list
     }
 
   let to_ml { main_module; aliases; shadowed } =
     let b = Buffer.create 16 in
     Buffer.add_string b "(* generated by dune *)\n";
-    let main_module = Module_name.to_string main_module in
+    let main_module = main_module |> Option.map ~f:Module_name.to_string in
```

Comment:
Note that this is a hack and we should be getting the proper canonical path. I'll fix it a subsequent PR.